File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        } 

        unordered_map<char, int> m;
        unordered_map<char, int> n;
        for (char c : s) m[c]++;
        for (char c : t) n[c]++;
            
        if (m == n) return true;
        
        return false;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> result = new HashMap<String, List<String>>();
        for (String str : strs) {
            isAnagram(result, str);
        }
        return new ArrayList<List<String>>(result.values());
    }

    public void isAnagram(HashMap<String, List<String>> result, String str) {
        char[] arr = new char[128];
        for (int i = 0; i < str.length(); i++) {
            arr[str.charAt(i)]++;
        }
        String answer = new String(arr);
        if (result.containsKey(answer)) result.get(answer).add(str);
        else {
            List<String> list = new ArrayList<String>();
            list.add(str);
            result.put(answer, list);
        }
    }
}
